tools/cutadapt/main.nf

```diff
@@ -5,44 +5,49 @@ nextflow.enable.dsl=2
 
 // Trimming reusable component
 process cutadapt {
-    publishDir "${params.outdir}/cutadapt",
-        mode: "copy", overwrite: true
+    publishDir "${params.outdir}/${opts.publish_dir}",
+    mode: "copy", 
+    overwrite: true,
+    saveAs: { filename ->
+                    if (opts.publish_results == "none") null
+                    else filename }
     
     container 'luslab/nf-modules-cutadapt:latest'
 
     input:
-        tuple val(sample_id), path(reads)
+        val opts
+        tuple val(meta), path(reads)
 
     output:
-        tuple val(sample_id), path("*.trimmed.fq.gz"), emit: trimmedReads
+        tuple val(meta), path("*.trimmed.fq.gz"), emit: trimmedReads
         path "*.txt", emit: report
 
     script:
-
-    // Check main args string exists and strip whitespace
-    args = ''
-    if(params.cutadapt_args && params.cutadapt_args != '') {
-        ext_args = params.cutadapt_args
-        args += " " + ext_args.trim()
-    }
-
-    // Construct CL line
-    cutadapt_command = "cutadapt${args} -o ${sample_id}.trimmed.fq.gz $reads > ${sample_id}_cutadapt.txt"
-
-    // Log
-    if (params.verbose){
-        println ("[MODULE] cutadapt command: " + cutadapt_command)
-    }
-
-    //SHELL
-    readList = reads.collect{it.toString()}
-    if (readList.size > 1){
-        """
-        cutadapt${args} -o ${reads[0].simpleName}.trimmed.fq.gz -p ${reads[1].simpleName}.trimmed.fq.gz $reads > ${sample_id}_cutadapt.txt
-        """
-    } else {
-        """
-        ${cutadapt_command}
-        """
-    }
+        args = ""
+        if(opts.args && opts.args != '') {
+            ext_args = opts.args
+            args += ext_args.trim()
+        }
+
+        prefix = opts.suffix ? "${meta.sample_id}${opts.suffix}" : "${meta.sample_id}"
+
+        // Construct CL line
+        cutadapt_command = "cutadapt ${args} -o ${meta.sample_id}.trimmed.fq.gz $reads > ${meta.sample_id}_cutadapt.txt"
+
+        // Log
+        if (params.verbose){
+            println ("[MODULE] cutadapt command: " + cutadapt_command)
+        }
+
+        //SHELL
+        readList = reads.collect{it.toString()}
+        if (readList.size > 1){
+            """
+            cutadapt ${args} -o ${reads[0].simpleName}.trimmed.fq.gz -p ${reads[1].simpleName}.trimmed.fq.gz $reads > ${meta.sample_id}_cutadapt.txt
```

Comment:
can you create two commands above and apply the same suffix to this? just to keep it looking nice - command_se and command_pe or something